### PR TITLE
added document index, type and id to crud operations

### PR DESCRIPTION
**Pull request by bartekbp**

### CODE REVIEW

File: jest-common/src/main/java/io/searchbox/client/JestClient.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
     <T extends JestResult> T execute(Action<T> clientRequest) throws IOException;
 
-    <T extends JestResult> void executeAsync(Action<T> clientRequest, JestResultHandler<T> jestResultHandler);
+    <T extends JestResult> void executeAsync(Action<T> clientRequest, JestResultHandler<? super T> jestResultHandler);
```
Comment (kramer) [inline jest-common/src/main/java/io/searchbox/client/JestClient.java]: What's the reason for this (as `T` is already defined as `T extends JestResult`)?

**Comment by kramer** [comment]
Could you also please add assertions for `getIndex`, `getType`, and `getId` on the result in the related actions' integration tests?

File: jest-common/src/main/java/io/searchbox/client/JestClient.java (hunk 1 above)
Comment (bartekbp) [inline jest-common/src/main/java/io/searchbox/client/JestClient.java]: That way we can use more generic handlers - e.g. JestResultHandler<JestResult> when executeAsync is parametrized with DocumentResult.
By introducing new type DocumentResult I wanted to keep currently working handlers intact.

Comment (kramer) [inline jest-common/src/main/java/io/searchbox/client/JestClient.java]: Oh I missed that, good point.

**Comment by bartekbp** [comment]
Done;)

**Comment by kramer** [comment]
Great, thanks for your contribution @bartekbp !

**Pull request merged**